model/publishing/delivery/tasks/RemoteDeliveryPublishingTask.php

```diff
@@ -0,0 +1,290 @@
+    private function getLoggerService(): LoggerService
+    {
+        return $this->getServiceLocator()->get(LoggerService::SERVICE_ID);
+    }
+
+    /**
+     * @return array
+     * @throws \core_kernel_persistence_Exception
+     */
+    private function prepareRequestData(): array
+    {
+        try {
+            $qtiPackageFile = $this->getQtiTestPackageFile($this->delivery);
+            $requestData = [
+                [
+                    'name' => RestTest::REST_FILE_NAME,
+                    'filename' => RestTest::REST_FILE_NAME,
+                    'contents' => $qtiPackageFile->readPsrStream(),
+                ],
+                [
+                    'name' => RestTest::REST_IMPORTER_ID,
+                    'contents' => 'taoQtiTest',
+                ],
+                [
+                    'name' => RestTest::REST_DELIVERY_PARAMS,
+                    'contents' => json_encode(
+                        [
+                            PublishingDeliveryService::ORIGIN_DELIVERY_ID_FIELD => $this->delivery->getUri()
+                        ]
+                    )
+                ]
+            ];
+
+            $deliveryClass = current($this->delivery->getTypes());
+            if ($deliveryClass->getUri() != DeliveryAssemblyService::CLASS_URI) {
+                $requestData[] = [
+                    'name' => RestTest::REST_DELIVERY_CLASS_LABEL,
+                    'contents' => $deliveryClass->getLabel()
+                ];
+            }
+
+            return $requestData;
+        } catch (FileNotFoundException $e) {
+            $this->getLoggerService()->logError($e->getMessage(), [$e->__toString()]);
+            $message = sprintf(__('QTI Test backup file not found for delivery "%s"'), $this->delivery->getLabel());
+            throw new PublishingFailedException($message);
+        }
+    }
+
+    /**
+     * @param array $requestData
+     * @return ResponseInterface
+     */
+    private function callRemotePublishingApi(array $requestData): ResponseInterface
+    {
+        try {
+            $body = new MultipartStream($requestData);
+            $request = new Request('POST', '/taoDeliveryRdf/RestTest/compileDeferred');
+            $request = $request->withBody($body);
+
+            return $this->getServiceLocator()->get(PlatformService::class)->callApi($this->environment->getUri(), $request);
+        } catch (ConnectException $e) {
+            $this->getLoggerService()->logError($e->getMessage(), [$e->__toString()]);
+
+            $message = __('Remote environment "%s" is not reachable.', $this->environment->getLabel());
+            throw new PublishingFailedException($message);
+        } catch (ClientException $e) {
+            $this->getLoggerService()->logError($e->getMessage(), [$e->__toString()]);
+
+            $response = $e->getResponse();
+            $message = __('Bad request.');
+            if ($response instanceof ResponseInterface && $response->getStatusCode() === 401) {
```

Comment:
`$response` will always be there for a `ClientException`.
`null` response usage was deprecated by Guzzle `6.3`.

Reply:
Removed this condition.